### ISSUE TITLE
--rpc-url seems to require a 443 port

### ISSUE THREAD

**Issue by tyvdh** [stellar/soroban-cli#255]
### What version are you using?
```
$ soroban version
soroban-cli 0.2.1 (2a065eeebc94f6c0b4fcb03a7963e139ff2a583d)
soroban-env-interface-version: 23
```

### What did you do?
```
soroban deploy --wasm target/wasm32-unknown-unknown/release/soroban_hello_world_contract.wasm --rpc-url https://kalepail-futurenet.stellar.quest/soroban/rpc
```
`--secret-key` & `--network-passphrase` have been set as `export` variables in the bash.

### What did you expect to see?
```
success
edc3f3077d4d07172392e9483682cc7f220bb7af30cd880ea7916414c2fbf58b
```

### What did you see instead?
```
error: jsonrpc error: Networking or low-level protocol error: Invalid Url: Port number is missing in the URL
```

**Comment by tyvdh** [comment]
Note if I use `--rpc-url https://kalepail-futurenet.stellar.quest/soroban/rpc:443` I get the expected result. However the `:443` port should not be required.

**Comment by Kanaye** [comment]
This is caused by this line in jsonrpsee: https://github.com/paritytech/jsonrpsee/blob/8622facbacee3c8f3616e489dca7498f77c54610/client/http-client/src/transport.rs#L67

And looks like they decided against default port numbers:
https://github.com/paritytech/jsonrpsee/issues/554

**Comment by niklasad1** [comment]
nah, it's blocked on https://github.com/hyperium/http/issues/509

**Comment by leighmcculloch** [comment]
I came across this bug and opened a fix for it. I missed we had an open issue for it already assigned to @tsachiherman. In any case there's a fix in #576.
